src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -570,6 +583,18 @@ public static void removeCode(TenantIdentifier tenantIdentifier, Storage storage
         });
     }
 
+    public static void removeDevice(TenantIdentifier tenantIdentifier, Storage storage,
+                                    String deviceIdHash)
+            throws StorageQueryException, StorageTransactionLogicException {
+        PasswordlessSQLStorage passwordlessStorage = StorageUtils.getPasswordlessStorage(storage);
+
+        passwordlessStorage.startTransaction(con -> {
+            passwordlessStorage.deleteDevice_Transaction(tenantIdentifier, con, deviceIdHash);
```

Comment:
this will delete the passwordless code and everything related to that device right? For example, if there are many codes created for the device (is that even possible?), it will delete all of them right?

Reply:
yes, they get deleted via FK cascade delete